x-pack/plugins/maps/public/classes/joins/inner_join.ts

```
@@ -52,12 +67,15 @@ export class InnerJoin {
     return `${this.getSourceDataRequestId()}_${FORMATTERS_DATA_REQUEST_ID_SUFFIX}`;
   }
 
-  getLeftField() {
-    return this._leftField;
+  getLeftField(): IField {
+    return this._leftField!;
```

Should this throw if left field is undefined?